app/controllers/apis/api_controller.rb

```diff
@@ -0,0 +1,43 @@
+# apis controller
+module Apis
+  class ApiController < ApplicationController
+    skip_before_action :authenticate_user!
+    protect_from_forgery with: :null_session
+    before_action :set_params_hash, if: :request_is_get?
+
+    def request_is_get?
+      request.method == "GET" && !params["params"].nil?
+    end
+
+    def request_is_valid_post?
+      request.method == "POST" && !except_params.nil?
+    end
+
+    def except_params
+      params.except(:action,
+                    :controller).empty?
+    end
+
+    def render_403(response)
+      render json: {
+        status: 403,
+        message: response.errors.messages
+      }
+    end
+
+    def mandatory_keys(method_name)
```

Comment:
It seems `method_name` is going to be list of param keys, the `method_name` name doesn't sound right.